### PR TITLE
feat(noUndeclaredEnvVars): add bracket notation and Bun.env support

### LINKED ISSUE #8494: 📎 Support other methods of accessing environment variables for `noUndeclaredEnvVars`
### Description

When using [noUndedclaredEnvVars](https://biomejs.dev/linter/rules/no-undeclared-env-vars/), it should also track/capture cases when using `Bun.env` and string accessed environment variables (i.e. `process.env["MY_VAR"]` instead of `process.env.MY_VAR`).

### PR DESCRIPTION
**Pull request by tt-a1i**
## Summary

Closes #8494

- Added support for bracket notation: `process.env["VAR"]`, `import.meta.env["VAR"]`
- Added support for Bun runtime: `Bun.env.VAR`, `Bun.env["VAR"]`
- Dynamic property access (variables, template literals with interpolation, concatenation) is still skipped

## Test plan

- Added test cases in `invalid.js` for bracket notation and Bun.env
- Added test cases in `valid.js` for allowed variables with bracket notation and Bun.env
- Updated `validDynamicAccess.js` to clarify that only truly dynamic access is skipped
- All 26 existing tests pass

## AI Assistance Disclosure

I used Codex to review the changes, sanity-check the implementation against existing patterns, and help spot potential edge cases.

### CODE REVIEW

File: .changeset/bracket-bun-env-support.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@biomejs/biome": patch
+---
+
+Extended [`noUndeclaredEnvVars`](https://biomejs.dev/linter/rules/no-undeclared-env-vars/) to support bracket notation (`process.env["VAR"]`, `import.meta.env["VAR"]`) and Bun runtime (`Bun.env.VAR`, `Bun.env["VAR"]`). Fixes [#8494](https://github.com/biomejs/biome/issues/8494).
```
Comment (ematipico) [inline .changeset/bracket-bun-env-support.md]:
```suggestion
Fixed [#8494](https://github.com/biomejs/biome/issues/8494). Extended [`noUndeclaredEnvVars`](https://biomejs.dev/linter/rules/no-undeclared-env-vars/) to support bracket notation (`process.env["VAR"]`, `import.meta.env["VAR"]`) and Bun runtime (`Bun.env.VAR`, `Bun.env["VAR"]`). 
```

We always frame the work with a past sentence. This is covered in the contribution guidelines

File: crates/biome_js_analyze/src/lint/nursery/no_undeclared_env_vars.rs [hunk 2]
```diff
@@ -88,11 +95,11 @@ declare_lint_rule! {
 /// State that holds the environment variable name being accessed
 pub struct EnvVarAccess {
     /// The name of the environment variable
-    env_var_name: TokenText,
+    env_var_name: String,
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_undeclared_env_vars.rs]: Any chance we can avoid storing a String? There's no need, If you can't extract `TokenText`, you can store a `HtmlSyntaxToken`. From there, we can read its trimmed value when we print the diagnostic

File: crates/biome_js_analyze/src/lint/nursery/no_undeclared_env_vars.rs [hunk 3]
```diff
@@ -163,17 +161,71 @@ impl Rule for NoUndeclaredEnvVars {
...
+    }
+
+    let parent_object = parent_member.object().ok()?;
+
+    // Get the env var name being accessed (e.g., NODE_ENV from process.env.NODE_ENV)
+    let member = static_expr.member().ok()?;
+    let env_var_name = member.as_js_name()?.value_token().ok()?;
+    let env_var_text = env_var_name.token_text_trimmed();
+
+    Some((env_var_text.to_string(), parent_object))
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_undeclared_env_vars.rs]: Here we can return the `value_token` so we don't need to store a string

File: crates/biome_js_analyze/src/lint/nursery/no_undeclared_env_vars.rs [hunk 4]
```diff
@@ -163,17 +161,71 @@ impl Rule for NoUndeclaredEnvVars {
...
+    let member = member.omit_parentheses();
+
+    // Check if it's a string literal
+    let string_literal = member
+        .as_any_js_literal_expression()?
+        .as_js_string_literal_expression()?;
+    let string_token = string_literal.value_token().ok()?;
+    let env_var_name = inner_string_text(&string_token);
+
+    Some((env_var_name.to_string(), parent_object))
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_undeclared_env_vars.rs]: Same here, we can return the `value_token`

File: crates/biome_js_analyze/src/lint/nursery/no_undeclared_env_vars.rs [hunk 5]
```diff
@@ -163,17 +161,71 @@ impl Rule for NoUndeclaredEnvVars {
...
+fn is_global_env_object(expr: &AnyJsExpression, model: &biome_js_semantic::SemanticModel) -> bool {
     let Some((reference, name)) = global_identifier(expr) else {
         return false;
     };
 
-    // Check that it's named "process" and not bound to a local variable
-    name.text() == "process" && model.binding(&reference).is_none()
+    // Check that it's named "process" or "Bun" and not bound to a local variable
+    let name_text = name.text();
+    (name_text == "process" || name_text == "Bun") && model.binding(&reference).is_none()
```
Comment (ematipico) [inline crates/biome_js_analyze/src/lint/nursery/no_undeclared_env_vars.rs:224]: We should add Deno too: https://docs.deno.com/runtime/reference/env_variables/

It feels "unfair" to just support Bun, when Deno is also right there.

**Review by ematipico — COMMENTED** [review]
I love the improvements, thank you @tt-a1i  for that. I left some feedback. Mainly: 
- add `Deno.env.get`
- remove the allocation of `String`

**Comment by tt-a1i** [comment]
> I love the improvements, thank you @tt-a1i for that. I left some feedback. Mainly:
> 
> * add `Deno.env.get`
> * remove the allocation of `String`

Done, also added Deno.env.get support with tests.

File: crates/biome_js_analyze/tests/specs/nursery/noUndeclaredEnvVars/invalid.js.snap [hunk 6]
```diff
@@ -70,6 +82,93 @@ invalid.js:9:23 lint/nursery/noUndeclaredEnvVars ━━━━━━━━━━
...
+```
+
+```
+invalid.js:20:17 lint/nursery/noUndeclaredEnvVars ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! The environment variable DENO_CUSTOM_VAR is not listed as a dependency in turbo.json. Add this environment variable a task's 'env'/'passThroughEnv', or to 'globalEnv', 'globalPassThroughEnv', in your turbo.json(c) configuration to ensure correct caching behavior in Turborepo.
+  
+    19 │ // Deno.env.get should also be checked
+  > 20 │ const denoVar = Deno.env.get("DENO_CUSTOM_VAR");
+       │                 ^^^^^^^^^^^^
```
Comment (ematipico) [inline crates/biome_js_analyze/tests/specs/nursery/noUndeclaredEnvVars/invalid.js.snap:155]: We could make the diagnostic text range consistent. This one isn't aligned with the others. Ideally, the diagnostic should highlight `DENO_CUSTOM_VAR`, because that's the issue. 

This isn't a blocker though, is if you want to improve it, feel free to send a PR :)

**Review by ematipico — APPROVED** [review]
